### ISSUE TITLE
Rack tests are non-deterministic

### ISSUE THREAD

**Issue by ahayworth** [open-telemetry/opentelemetry-ruby-contrib#387]
**Description of the bug**

Rack tests are non-deterministic. It is trivial to reproduce -

Good run:
```
opentelemetry-ruby-contrib/instrumentation/rack on  ahayworth/rack-test-ordering-failure [?] via 💎 v3.0.4 took 2s
zsh ❯ bundle exec rake test SEED=1
/Users/andrew/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/yard-doctest-0.1.17/lib/yard-doctest.rb:57: warning: `&' interpreted as argument prefix
Run options: --seed 1

# Running:

.................................................

Finished in 0.014089s, 3477.8905 runs/s, 6813.8262 assertions/s.

49 runs, 96 assertions, 0 failures, 0 errors, 0 skips
```

Bad run:

```

opentelemetry-ruby-contrib/instrumentation/rack on  main via 💎 v3.0.4
zsh ❯ bundle exec rake test SEED=6372
/Users/andrew/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/yard-doctest-0.1.17/lib/yard-doctest.rb:57: warning: `&' interpreted as argument prefix
Run options: --seed 6372


# Running:

F................................................

Finished in 0.015968s, 3068.6373 runs/s, 5323.1463 assertions/s.

  1) Failure:
OpenTelemetry::Instrumentation::Rack::Instrumentation::when rack gem does not exist#test_0001_skips installation [/Users/andrew/projects/opentelemetry-ruby-contrib/instrumentation/rack/test/opentelemetry/instrumentation/rack/instrumentation_test.rb:45]:
Expected nil (NilClass) to respond to #empty?.

49 runs, 85 assertions, 1 failures, 0 errors, 0 skips
rake aborted!
Command failed with status (1)
/Users/andrew/.rbenv/versions/3.0.4/bin/bundle:23:in `load'
/Users/andrew/.rbenv/versions/3.0.4/bin/bundle:23:in `<main>'
Tasks: TOP => test
(See full trace by running task with --trace)
```

It seems as if the tests in `test/opentelemetry/instrumentation/rack/instrumentation_test.rb` are run _first_, then the instrumentation's `config` hash is always empty (and thus further assertions on the hash keys fail). It seems to be related to how we are resetting the instrumentation in that test with an explicitly empty config hash - the defaults aren't getting loaded as we expect.

cc @arielvalentin - I think this change started failing after #356

**Comment by ahayworth** [comment]
Oh, whoops this is a duplicate of #372 🤦
